app/services/image_filename_service.rb

```diff
@@ -1,15 +1,22 @@
 # frozen_string_literal: true
 
 class ImageFilenameService
+  MAX_LENGTH = 65
+
   attr_reader :document
 
   def initialize(document)
     @document = document
   end
 
   def call(suggested_name, mime_type)
-    normalised_name = normalise(suggested_name, mime_type)
-    ensure_unique(normalised_name)
+    filename = ActiveStorage::Filename.new(suggested_name)
+
+    base = filename.base.parameterize.slice 0...MAX_LENGTH
```

Comment:
I would split out the code to truncate the filename into separate commit, to make it clear what was changed because of truncation, and what has just been refactored, for example, method renames.

Reply:
As the service is new in this PR, I've squashed the changes into the original commit instead, to make it easier for people in future.